### PR TITLE
TEST: Re-enable GPU tests for decision trees

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

This PR re-enables tests for decision trees in GPUs which are currently deactivated and which should be solved with this PR in oneDAL: https://github.com/oneapi-src/oneDAL/pull/2984

~It also requires a version number bump: https://github.com/oneapi-src/oneDAL/pull/2981~

~Right now it shouldn't do anything, because those two PRs in oneDAL haven't been merged.~

All PR dependencies have been merged by now.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.

**Performance**

Not applicable.

### CODE REVIEW

File: sklearnex/ensemble/tests/test_forest.py [hunk 1]
```diff
@@ -63,8 +63,8 @@ def test_sklearnex_import_rf_classifier(dataframe, queue, block, trees, rows, sc
 
 @pytest.mark.parametrize("dataframe,queue", get_dataframes_and_queues())
 def test_sklearnex_import_rf_regression(dataframe, queue):
-    if queue and queue.sycl_device.is_gpu:
-        pytest.skip("RF regressor predict for the GPU sycl_queue is buggy.")
+    if (not daal_check_version((2025, "P", 200))) and queue and queue.sycl_device.is_gpu:
+        pytest.skip("Bug fixed in a later version.")
```
Comment (icfaust) [inline sklearnex/ensemble/tests/test_forest.py:67]: id say what the bug was

**Review by icfaust — COMMENTED** [review]
just a small request in terms of understanding the skip, beyond that ready to go. i want to see green CIs, and when out of draft i will approve

File: sklearnex/ensemble/tests/test_forest.py (hunk 1 above)
Comment (david-cortes-intel) [reply sklearnex/ensemble/tests/test_forest.py:67]: Do you mean to have the issue described in the skip message?

**Comment by david-cortes-intel** [comment]
Looks like this fix uncovered a different error now that trees are non-empty:
```
BaseClassifier in AdaBoostClassifier ensemble is worse than random, ensemble can not be fit.
```

Will look into it.

**Comment by david-cortes-intel** [comment]
On a deeper look, the error about "ensemble is worse than random" appears to happen only on CPU, only on Windows, and only on python3.9, so it shouldn't be anyhow connected to this PR.

What's worse, it doesn't use random seeds for randomized algorithms so the error is not going to be easily reproducible.

I'm unable to reproduce it on linux, and given that it only happens in an old python version, I've added it to the deselections. It wasn't clear to me if they allow de-selecting for a specific python version, so I disabled it for windows+sklearn>=1.4.

**Comment by david-cortes-intel** [comment]
CI is passing now, ready for review.

File: sklearnex/ensemble/tests/test_forest.py (hunk 1 above)
Comment (icfaust) [reply sklearnex/ensemble/tests/test_forest.py:67]: sorry about the delay. you are exactly right

Comment (david-cortes-intel) [reply sklearnex/ensemble/tests/test_forest.py]: Added a comment about the bug and version to the skip message.

**Review by icfaust — APPROVED** [review]
two others still remain which would be nice to have a similar skip message. when changed can merge

**Comment by david-cortes-intel** [comment]
> two others still remain which would be nice to have a similar skip message. when changed can merge

Updated the message in the other skip reasons.
